redash/tasks/worker.py

```diff
@@ -3,13 +3,21 @@
 import signal
 import time
 from redash import statsd_client
+from sys import platform
 from rq import Queue as BaseQueue, get_current_job
+from rq import Worker as BaseWorker
 from rq.worker import HerokuWorker # HerokuWorker implements graceful shutdown on SIGTERM
 from rq.utils import utcnow
 from rq.timeouts import UnixSignalDeathPenalty, HorseMonitorTimeoutException
 from rq.job import Job as BaseJob, JobStatus
 
-
+# HerokuWorker does not work in OSX https://github.com/getredash/redash/issues/5413
+def get_worker_base_class():
+    if platform == "darwin":
+        return BaseWorker
+    else:
+        return HerokuWorker
```

Comment:
A more idiomatic way would be:

```python
if platform == "darwin":
    Worker = BaseWorker
else:
    Worker = HerokuWorker
```

And then subclass from `Worker` where you used `get_worker_base_class` before.